### PR TITLE
Adding offloading applet

### PR DESCRIPTION
**Pull request by danielmgit**
Hints:
- Adding offloading applet
- Removing usage of non-used applet arguments
- Small fixes

### CODE REVIEW

File: applets/006-tcpdump/applet.py [hunk 1]
```diff
@@ -139,8 +140,9 @@ def tcpdump_stop(x, arg_d):
 
 def main(x, conf, args):
     if not len(args) >= 4:
-        x.p.msg("wrong usage. use: [hostname] id:[number] mode:[start|stop] "
-                "local-file-name:\"[filename]\" filter:\"[filter]\"\n")
+        x.p.err("wrong usage. use: [hostname] id:[number] mode:[start|stop]\n"
+                "mode:start -> \tfilter:\"[filter]\"\n"
```
Comment (hgn) [inline applets/006-tcpdump/applet.py:146]: can filter be empty?

File: applets/105-offloading/applet.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#  - exec-applet 105-offloading alpha command:"tso off ufo off gso off gro off
+#    lro off"
+#
+# Hints:
+#  - "Cannot change [...]" messages probably mean that the value can not be
+#    changed (is fixed)
+
+
+def print_usage(x):
+
```
Comment (hgn) [inline applets/105-offloading/applet.py]: no newline here.

See https://github.com/danielmgit/net-applet-shuffler/blob/master/nas.py - there is never a newline after the function header.

File: applets/105-offloading/applet.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    x.p.msg("\n usage:\n", False)
+    x.p.msg(" - exec 105-offloading [host] command:\"[command]\"\n", False)
+    x.p.msg("\n check success:\n", False)
+    x.p.msg(" - ethtool -k [interface]\n", False)
+
+    return False
```
Comment (hgn) [inline applets/105-offloading/applet.py]: **print_usage()** always success - no need for a return value

File: applets/105-offloading/applet.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def set_offloading(x, conf, dic):
+
```
Comment (hgn) [inline applets/105-offloading/applet.py]: remove newline

File: applets/105-offloading/applet.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            x.p.err("error: offloading could not be set for host {} device {}\n"
+                    .format(dic["host"], device))
+            x.p.err("failed cmd: \"{}\"".format(cmd))
+            return False
+
+    return True
+
+
+def main(x, conf, args):
+
```
Comment (hgn) [inline applets/105-offloading/applet.py]: no newline

File: campaigns/001-netperf-data-exchange/run.py [hunk 6]
```diff
@@ -66,6 +58,20 @@ def main(x):
     x.exec('102-tcp-route-metrics-save alpha route-metrics-save:disabled')
     x.exec('102-tcp-route-metrics-save beta route-metrics-save:disabled')
 
+    # interface offloading
+    # usage: [host] command:"[command]"
+    # restoration has to be covered manually (see end of run)
+    x.exec('105-offloading alpha command:"tso off ufo off gso off gro off lro off"')
```
Comment (hgn) [inline campaigns/001-netperf-data-exchange/run.py]: mmh, the syntax differs from all other applets - not really consistent. A except a) or b):
- tso:[on | off], gro:[on | off], ...
- offloading:[on | off]

Mhhhh

File: nas.py [hunk 7]
```diff
@@ -192,16 +192,21 @@ def get_middle_box_iface_name_by_network_name(self, host_name, network):
               "host {} and network name {}\n".format(host_name, network))
         sys.exit(1)
 
-    def get_all_iface_names(self, host_name):
+    def get_all_test_iface_names(self, host_name):
```
Comment (hgn) [inline nas.py]: the naming is a little bit strange. I know a differentiation between data and control (e.g. data plane, control plane). I think we should substitute every test with data. Simple search and replace ...

File: applets/006-tcpdump/applet.py (hunk 1 above)
Comment (danielmgit) [inline applets/006-tcpdump/applet.py:146]: Good question. In fact, filter CAN be empty, and EVERYTHING on the interface will be capped.

File: applets/105-offloading/applet.py (hunk 5 above)
Comment (danielmgit) [inline applets/105-offloading/applet.py]: All the precious newlines will be removed.

File: campaigns/001-netperf-data-exchange/run.py (hunk 6 above)
Comment (danielmgit) [inline campaigns/001-netperf-data-exchange/run.py]: b) will be done.

File: nas.py (hunk 7 above)
Comment (danielmgit) [inline nas.py]: Fine.

File: applets/105-offloading/applet.py (hunk 3 above)
Comment (danielmgit) [inline applets/105-offloading/applet.py]: It was used the following way:
if "?" in args:
    return print_usage(x)

Changed it now.

**Comment by danielmgit** [comment]
Comments processed. Next merge request.

File: applets/006-tcpdump/applet.py (hunk 1 above)
Comment (hgn) [inline applets/006-tcpdump/applet.py]: Great!

**Pull request merged**